src/test/java/com/google/sps/servlets/UserDataServletTest.java

```diff
@@ -0,0 +1,256 @@
+  // Uses a local datastore stored in memory for tests
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private UserDataServlet servletUnderTest;
+  private StringWriter stringWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new UserDataServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the doGet method, where the datastore doesn't contain the test user id.
+   * Expected output: JSON response with "user-found" property as false.
+   */
+  @Test
+  public void testGetMethodWithInvalidId() throws Exception {
+    // Mock the getParameter call
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+
+    // Store output given by mockResponse
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+    when(mockResponse.getContentType()).thenReturn("application/json");
+
+    // Create a local datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // Test the doGet method by calling the wrapper method
+    servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
+    writer.flush();
+
+    // Build the correct response output
+    Map<String, Object> correctResponse = new HashMap<>();
+    correctResponse.put(USER_FOUND_PROPERTY, false);
+
+    // Check that the user wasn't found in datastore
+    verify(mockRequest, atLeast(1)).getParameter(USER_ID_PROPERTY);
+    Assert.assertEquals(gson.toJson(correctResponse), stringWriter.toString().trim());
+  }
+
+  /**
+   * Tests the doGet method, where datastore does contain the test user id.
+   * Expected output: JSON response with "user-found" property as true,
+   * and the user properties filled in with the test user properties.
+   */
+  @Test
+  public void testGetMethodWithValidId() throws Exception {
+    // Mock the getParameter call
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+
+    // Store output given by mockResponse
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+    when(mockResponse.getContentType()).thenReturn("application/json");
+
+    // Create a local datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    addTestUserEntityToDatastore(datastore);
+    // Test the doGet method by calling the wrapper method
+    servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
+    writer.flush();
+
+    // Build the correct response output
+    Map<String, Object> correctResponse = new HashMap<>();
```

Comment:
ImmutableMap<String, Object> expected = ImmutableMap.of(
    USER_FOUND_PROPERTY, true,
    USER_NAME_PROPERTY, TEST_USER_NAME,
    ...
);

Reply:
I used an `ImmutableMap`, but used the `put` method for each property. The static `of` method didn't compile, the documentation [here](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableMap.html#of--) details that of takes a maximum of 6 key value pairs.